### PR TITLE
Feature/database performance audit

### PR DESCRIPTION
**Pull request by Salimer**
## The following PR is raised to implement the following tasks:

The following queries are taking too much time (1 sec = 1000ms can be considered as too much time for database query). Try them on your machine to confirm it:

`EXPLAIN ANALYZE SELECT COUNT(*) FROM visits where animal_id = 4;`
Screenshot 2023-07-17 at 16 51 11

`EXPLAIN ANALYZE SELECT * FROM visits where vet_id = 2;`
Screenshot 2023-07-17 at 16 51 45

`EXPLAIN ANALYZE SELECT * FROM owners where email = 'owner_18327@mail.com';`
Screenshot 2023-07-17 at 16 52 05

- [x] Find a way to decrease the execution time of the first query. Look for hints in the previous lessons.

- [x] Find a way to improve execution time of the other two queries.

- [x] While you are making changes, check if they help by running EXPLAIN ANALYZE. Once you succeed, take a screenshot of the EXPLAIN ANALYZE result showing that time actually decreased.

After creating indexes to improve performance:

`EXPLAIN ANALYZE SELECT COUNT(*) FROM visits where animal_id = 4;`
Screenshot 2023-07-17 at 17 14 08

`EXPLAIN ANALYZE SELECT * FROM visits where vet_id = 2;`
Screenshot 2023-07-17 at 18 17 37

`EXPLAIN ANALYZE SELECT * FROM owners where email = 'owner_18327@mail.com';`
Screenshot 2023-07-17 at 17 31 26

- [x] Any changes you made to your database schema should be added to the schema.sql file and commited in a new pull request.

- [x] If you decide to commit any INSERT INTO queries - remember to add them to the data.sql file. It is important to keep only the queries that change the database structure in the schema.sql file.

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -89,4 +89,10 @@ CREATE TABLE
 
 vet_clinic=# ALTER TABLE visits
 vet_clinic-# ADD COLUMN visit_date DATE;
-ALTER TABLE
\ No newline at end of file
+ALTER TABLE
+
+vet_clinic=# CREATE INDEX idx_animal_id ON visits (animal_id);
+CREATE INDEX
```
Comment (souad988) [inline schema.sql]: Please add a valid query as your schema should contain queries that can be executed 
`CREATE INDEX idx_visits_animal_id ON visits (animal_id);`

File: schema.sql [hunk 2]
```diff
@@ -89,4 +89,10 @@ CREATE TABLE
...
+
```
Comment (souad988) [inline schema.sql]: the index for the second query `SELECT * FROM visits where vet_id = 2;` is missing kindly add it `CREATE INDEX idx_visits_vet_id ON visits (vet_id);`

also make sure that the the query planner output includes index scan as I can see only seq scan in both screenShots

**Review by souad988 — CHANGES_REQUESTED** [review]
Hi @Salimer , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

242722065-db428b0c-8dfc-49c4-a156-42f227e32868


## Highlights
Good collaboration 
Use of indexes for performance improvement


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Salimer** [comment]
Thank you @souad988 for the thorough review, we addressed all of the required changes

**Review by IngDavidTM — APPROVED** [review]
Hi @Salimer 🙋‍♂️, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**